annotation-processing/src/main/java/com/exactpro/epfast/annotation/processing/FastFieldResolver.java

```diff
@@ -23,7 +23,31 @@ public FastFieldResolver(Elements elements) {
     private boolean isOverriddenIn(FastTypeElement fastType, FastFieldElement fastField) {
         List<FastFieldElement> fastFieldElements = getFastFieldsFor(fastType.getElement());
         return fastFieldElements.stream().anyMatch(field -> field.getMethodName().equals(fastField.getMethodName()) &&
-            field.getParameterType().equals(fastField.getParameterType()));
+            haveValidParameterTypes(field.getParameterType(), fastField.getParameterType()));
+    }
+
+    private boolean haveValidParameterTypes(TypeMirror aParameter, TypeMirror bParameter) {
+        if (aParameter.equals(bParameter)) {
+            return true;
+        }
+        Element aClass = elements.getTypeElement(aParameter.toString());
+        Element bClass = elements.getTypeElement(bParameter.toString());
+        if (!isAInstanceOfB(aClass, bClass) && !isAInstanceOfB(bClass, aClass)) {
```

Comment:
We should act the same way as Java does. If we see a method in a base class, which is overridden by `@FastField` annotated method we've seen already (in derived class), then we should skip it. Otherwise we should add it to collection of fast fields.
```java
class A {
@FastField("xxx")
void setFoo(int x);
}

@FastType
class B extends A {
@FastField("yyy")
void setFoo(long x);
}
```
should be processed without errors and result in fields named `xxx` and `yyy` in `B`. If the names are same, we should output duplicate names error.
There is another edge case if `setFoo()` is overridden in derived class and both have `@FastField` annotations with different names. I don't think we should treat it as error, but we should warn about it.